File: Stack/stack.java
package Stack;
import java.util.EmptyStackException;

class Stack {
    private int[] arr;
    private int top;
    private int maxSize;

    public Stack(int maxSize) {
        this.arr = new int[maxSize];
        this.top = -1;
        this.maxSize = maxSize;
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == maxSize - 1;
    }

    public void push(int data) {
        if (isFull()) {
            System.out.println("Stack Overflow!");
            return;
        }
        arr[++top] = data;
    }

    public int pop() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return arr[top--];
    }

    public int peek() {
        if (isEmpty()) {
            throw new EmptyStackException();
        }
        return arr[top];
    }
}

public class Main {
    public static void main(String[] args) {
        Stack stack = new Stack(100);
        stack.push(1);
        stack.push(2);
        stack.push(3);
        System.out.println("Top element: " + stack.peek());
        System.out.println("Popped element: " + stack.pop());
        System.out.println("Top element: " + stack.peek());
    }
}

File: Stack/stack.kt
import java.util.EmptyStackException

class Stack(private val maxSize: Int) {
    private val arr = IntArray(maxSize)
    private var top = -1

    fun isEmpty(): Boolean {
        return top == -1
    }

    fun isFull(): Boolean {
        return top == maxSize - 1
    }

    fun push(data: Int) {
        if (isFull()) {
            println("Stack Overflow!")
            return
        }
        arr[++top] = data
    }

    fun pop(): Int {
        if (isEmpty()) {
            throw EmptyStackException()
        }
        return arr[top--]
    }

    fun peek(): Int {
        if (isEmpty()) {
            throw EmptyStackException()
        }
        return arr[top]
    }
}

fun main() {
    val stack = Stack(100)
    stack.push(1)
    stack.push(2)
    stack.push(3)
    println("Top element: ${stack.peek()}")
    println("Popped element: ${stack.pop()}")
    println("Top element: ${stack.peek()}")
}
